misc/kickstart_create_network_line.py

```diff
@@ -14,60 +14,56 @@
 #
 
 
-''' 
+'''
     This reads a configuration file and checks for functioning
-    network links in /sys/class/net/*, then emits a ks.cfg network line. 
+    network links in /sys/class/net/*, then emits a ks.cfg network line.
     '''
 
 import os
 import re
 
-# Global bad: 
-global DEBUG
-DEBUG = True
-
 global TO_LOG
 # This "logs" to stdout which is captured during kickstart
 TO_LOG = True
 
-# This is the standard interface we install to. It is set to a speed value of 
-# 5 (vice 100,1000 or 10000) later on and any other interface will override it 
-# if you've got something faster installed. 
+# This is the standard interface we install to. It is set to a speed value of
+# 5 (vice 100,1000 or 10000) later on and any other interface will override it
+# if you've got something faster installed.
 standard_interface=['p4p1']
 
 ## These are configuration settings:
 # Name of Configuration file:
 cfg_file = "network.cfg"
 
-# Where linux is putting the interface stuff: 
+# Where linux is putting the interface stuff:
 iface_dir = '/sys/class/net/'
 
 ignore_interfaces = ['lo','bond']
 
 # Where we kickstart mounts the ISO, and our config directory:
 base_cfg_dir = '/mnt/stage2/ks_scripts/'
 
-# Remember the ? makes the match non-greedy. This is important. 
+# Remember the ? makes the match non-greedy. This is important.
 cfg_line = re.compile("\s*(?P<key>.*?)=(?P<value>.*)\s*$")
 
-# Pick the interface speed for bonding, or "Auto". 
+# Pick the interface speed for bonding, or "Auto".
 # Auto assumes you want the fastest connections with more than 1 interface,
 # Or if there's not 2 interfaces at the same speed you want the fastest.
-# auto is expected to be a string, otherwise use integers: 
-# Speed is in megs. 1000 is 1 gig, 10000 is 10G. 
+# auto is expected to be a string, otherwise use integers:
+# Speed is in megs. 1000 is 1 gig, 10000 is 10G.
 
 iface_speed = 'auto'
 
-restring="%(iface_dir)s(?P<iface>.*)/speed" %vars()
+restring="".join([iface_dir, "(?P<iface>.*)/speed"])
```

Comment:
I wouldn't really say string joining like this is a "best practice". It arguably makes the statement harder to read and more error prone. I do agree with not using `vars()` though, but that can be changed like this: 
```
restring="%s(?P<iface>.*)/speed" % iface_dir
```
or
```
restring = iface_dir + "(?P<iface>.*)/speed"
```
What do you think? If 3.6 becomes the new default, I'd recommend using f-strings.

Ditto this comment to the other changes like this below.

Reply:
I remember reading somewhere that the recommended method of string concatenation was to use `join`. I just looked it up, though, and it appears that what I was thinking of was this line from PEP8:

> _" In performance sensitive parts of the library, the ''.join() form should be used instead. This will ensure that concatenation occurs in linear time across various implementations."_

So it's probably fine to use `+`/`+=` here since that's not exactly _"performance sensitive"_.

As to "fstring"s, I'd rather not break 2.7 compatibility immediately - though technically it should be fine because 3.x includes a deprecation notice for Python 2. I don't _think_ 2.7 has those.

Reply:
Yeah 2.7 definitely doesn't have f-strings. I'm just saying if/when 3.6 becomes the new default on centOS, we could/should use f-strings. But they should be disregarded for now since IMO this seems like something that depends on the default python installation which is still 2.7 AFAIK. So probably best to maintain 2/3 compatibility for the time being (otherwise it will probably break people).